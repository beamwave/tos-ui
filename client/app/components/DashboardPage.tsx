import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import ReactGridGallery from 'react-grid-gallery'
import { FontAwesomeIcon } from '../../../node_modules/@fortawesome/react-fontawesome'
import { check, deepEquals } from '../helpers/helpers'
import { startSearch } from '../actions/school'
import { throttle, debounce } from 'throttle-debounce'

// gradeAdjustmentStartDt = invoice start date

// invoice button display requirements: tests are shipped (not pending), correct invoice window, otherwise show invoice

// allow edit if invoice window still opened

// react acl

// public search is possible from any react page

// student names, prevent invoice generation

// fee waiver from order 30 (fee waivers available)

// data for students taking current test

// yearly changes to names, ie []

// order table: aicode (partition?), testDt, eventid (sort?), weborderid (index?), { lines } ?,

// get orders by aicode and possibly filter by year

// learn about aws playground

// get sample data for hyperloop events (orders, invoice, user)

// current possible tables: misgrid adjustments, orders, test takers, ship confirmation, reference (test date, window dates, event ids)

// add lambda tag: key: 'PERSISTENT': 'YES'

interface IProps {
  startSearch: (data: any) => any[]
  results: any
  // username: string
  // category: string
  // categories?: any
  // images: any[]
  // filters?: any[]
  // filter?: (by: FData) => any
  // history?: any
}

interface IState {
  // images: any[]
}

export class DashboardPage extends Component<IProps> {
  state = {
    state: 'all',
    filterType: 'school',
    stateCollege: 'bard college',
    districtCollege: 'bard college',
    stateContractCoverage: 'all',
    districtContractCoverage: 'state',
    schoolQuery: { active: true, query: '' },
    districtQuery: { active: false, query: '' },
    results: []
  }

  // componentWillReceiveProps = (nextProps, nextState) => {
  //   const { results } = nextProps
  //   this.setState({ results })
  // }

  onFieldChange = ({ target: { name, value } }) =>
    this.setState({ [name]: value } as any)

  onInputChange = ({ target: { name, value } }) =>
    this.setState({ [name]: { active: true, query: value } } as any)

  onFilterChange = ({ target: { name, value } }) => {
    // set filterType value
    this.setState({ [name]: value } as any)

    // set active state of queries
    if (value === 'school')
      this.state.districtQuery = { active: false, query: '' }

    if (value === 'district')
      this.state.schoolQuery = { active: false, query: '' }
  }

  onQuery = async userInput => this.props.startSearch(userInput)

  // call action creator 1 second after user finishes typing
  startSearchDebounce = debounce(1000, this.onQuery)

  // call action creator every 750 milliseconds during typing
  startSearchThrottle = throttle(750, this.onQuery)

  // store user input in local state ??
  onUserInput = userInput => this.setState({ userInput })

  // query db when user begins searching
  onSubmit = async e => {
    e.preventDefault()
    // const {
    //   target: { value: userInput }
    // } = e
    const { filterType } = this.state

    const data = {
      userInput: '',
      filters: {}
    }

    if (filterType === 'school') {
      const {
        schoolQuery: { query }
      } = this.state
      data.userInput = query
    }
    if (filterType === 'district') {
      const {
        districtQuery: { query },
        state,
        districtCollege,
        districtContractCoverage
      } = this.state
      data.userInput = query
      data.filters = { state, districtCollege, districtContractCoverage }
    }
    if (filterType === 'state') {
      const { stateCollege, stateContractCoverage } = this.state
      data.filters = { stateCollege, stateContractCoverage }
    }

    const { userInput } = data

    // start querying db at various intervals
    if (userInput.length < 5) this.startSearchThrottle(data)
    else this.startSearchDebounce(data)
  }

  displaySchoolInput = () => (
    <form onSubmit={this.onSubmit} className="school-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        onChange={this.onFilterChange}
        value={this.state.filterType}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>
      <input
        type="text"
        name="schoolQuery"
        className="searchbar-input"
        value={this.state.schoolQuery.query}
        onChange={this.onInputChange}
        placeholder="search school by name or code"
      />
      <button className="blue-button">Search</button>

      <p className="link">Advanced Search</p>
    </form>
  )

  displayDistrictInput = () => (
    <form onSubmit={this.onSubmit} className="district-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        value={this.state.filterType}
        onChange={this.onFilterChange}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>
      <input
        type="text"
        name="districtQuery"
        className="searchbar-input"
        value={this.state.districtQuery.query}
        onChange={this.onInputChange}
        placeholder="District Name, Code"
      />

      <p>in</p>
      <select
        name="state"
        className="searchbar-select"
        value={this.state.state}
        onChange={this.onFieldChange}
      >
        <option value="all">All</option>
        <option value="aa">AA</option>
        <option value="ae">AE</option>
        <option value="alabama">Alabama</option>
      </select>

      <select
        name="districtCollege"
        className="searchbar-select"
        value={this.state.districtCollege}
        onChange={this.onFieldChange}
      >
        <option value="bard college">Bard College</option>
        <option value="furman university">Furman University</option>
        <option value="harvard college">Harvard College</option>
        <option value="mercer university">Mercer University</option>
      </select>

      <p>Contract Coverage</p>
      <select
        name="stateContractCoverage"
        className="searchbar-select"
        value={this.state.stateContractCoverage}
        onChange={this.onFieldChange}
      >
        <option value="all">All</option>
        <option value="district">District</option>
        <option value="state">State</option>
        <option value="none">None</option>
      </select>

      <button className="blue-button">Search</button>
    </form>
  )

  displayStateInput = () => (
    <form onSubmit={this.onSubmit} className="state-input-container">
      <select
        name="filterType"
        className="searchbar-select"
        onChange={this.onFieldChange}
        value={this.state.filterType}
      >
        <option value="school">school</option>
        <option value="district">district</option>
        <option value="state">state</option>
      </select>

      <select
        name="stateCollege"
        className="searchbar-select"
        onChange={this.onFieldChange}
        value={this.state.stateCollege}
      >
        <option value="bards college">Bards College</option>
        <option value="charles r drew">
          Charles R. Drew University of Medicine and Science
        </option>
        <option value="dod">Department of Defense Education Activity</option>
      </select>

      <p>Contract Coverage</p>
      <select
        name="districtContractCoverage"
        className="searchbar-select"
        value={this.state.districtContractCoverage}
        onChange={this.onFieldChange}
      >
        <option value="state">State</option>
      </select>

      <button className="blue-button">View</button>
    </form>
  )

  render = () => {
    const { filterType } = this.state
    const { results } = this.props

    return (
      <div className="dashboard-container">
        <nav>
          <h2>Test Ordering</h2>
          <div>
            <p>Org Search</p>
            <p>Calendar</p>
            <p>Help</p>
          </div>
        </nav>
        <header>
          {filterType === 'school' && this.displaySchoolInput()}
          {filterType === 'district' && this.displayDistrictInput()}
          {filterType === 'state' && this.displayStateInput()}
        </header>
        <main>
          {results.length > 0 && (
            <h3>
              {results.length} Search Results for "{results[0].query}"
            </h3>
          )}
          <div className="results-table">
            {results &&
              results.map(
                ({
                  aicode,
                  schoolName,
                  schoolAddress,
                  schoolStatus,
                  schoolAdmins
                }) => [
                  <header key={`${aicode}-${schoolAddress}`}>
                    <p>School Name</p>
                    <p>Address</p>
                    <p>Status</p>
                    <p>Adminstration</p>
                    <div>
                      <p>Contract</p>
                      <div className="subhead">
                        <p>State</p>
                        <p>Distr.</p>
                      </div>
                    </div>
                    <p>Study</p>
                    <p>Research Group</p>
                  </header>,
                  <main key={`${aicode}-${schoolName}`}>
                    <div>
                      <p>{schoolName}</p>
                      <p>AI {aicode}</p>
                    </div>
                    <div>
                      <p>{schoolAddress}</p>
                    </div>
                    <div>
                      <p>{schoolStatus}</p>
                    </div>
                    <div className="admins">
                      {schoolAdmins.map(
                        ({ admin, date, contract, study, research }) => (
                          <div className="admin-row" key={`${admin}-${date}`}>
                            <div>
                              <p>{admin}</p>
                              <p className="date">{date}</p>
                            </div>
                            <p>{contract}</p>
                            <p>{study}</p>
                            <p>{research}</p>
                          </div>
                        )
                      )}
                    </div>
                    <div className="view-link">
                      <Link to="/school-landing">View</Link>
                    </div>
                  </main>

                ]
              )}
          </div>
          {/* <table>
            {results &&
              results.map(
                ({
                  aicode,
                  schoolName,
                  schoolAddress,
                  schoolStatus,
                  schoolAdmins
                }) => [
                  <thead>
                    <tr>
                      <th>School Name</th>
                      <th>Address</th>
                      <th>Status</th>
                      <th>Adminstration</th>
                      <th>Contract</th>
                      <th>Study</th>
                      <th>Research Group</th>
                    </tr>
                  <thead>,
                  <tbody>
                    <tr>
                      <td>
                        <tr>{schoolName}</tr>
                        <tr>AI {aicode}</tr>
                      </td>
                      <td>{schoolAddress}</td>
                      <td>{schoolStatus}</td>
                      <td colSpan={3}>
                        {schoolAdmins.map(
                          ({ admin, date, contract, study, research }) => [
                            <tr>
                              <td>
                                {admin}
                                {'\n'}
                                {date}
                              </td>
                              <td>{contract}</td>
                              <td>{study}</td>
                              <td>{research}</td>
                            </tr>
                          ]
                        )}
                      </td>
                    </tr>
                  </tbody>
                ]
              )}
          </table> */}
        </main>
      </div>
    )
  }
}

// aicode: action.results.aicode,
// schoolName: action.results.schoolName,
// schoolAddress: action.results.schoolAddress,
// schoolStatus: action.results.schoolStatus,
// schoolAdmins: action.results.schoolAdmins

const mapStateToProps = (state, props) => ({
  results: state.school.results
  // token: state.auth.token,
  // username: state.auth.username,
  // category: state.app.category,
  // categories: state.images.categories,
  // images: state.images.all,
  // filters: state.filters
})

export default connect<any>(
  mapStateToProps,
  { startSearch }, // enables use of startSearch action creator
  null,
  { pure: false }
)(DashboardPage)

// state = {
//   images: [],
//   categories: 'map an object'
// }

// filter images
// handleFilterChange = ({ target }) => {
//   const { filters, filter } = this.props
//   // "medium", "device", "wardrobe", "mood", etc.
//   // console.log('dataset', target.dataset.filterName)
//   // console.log('value', target.value)

//   const by = {
//     category: target.dataset.filterName,
//     value: target.value
//   }

//   filter(by)
// }

// handleThumbnailClick = (i, e) => {
//   const { images, history } = this.props
//   const imageId = images[i]._id
//   history.push(`/image/${imageId}`)
// }

// componentDidMount = () => {
//   const { images: imagesFromServer } = this.props
//   // console.log('imagesFromServer', imagesFromServer)
//   let images
//   if (check(imagesFromServer) && Array.isArray(imagesFromServer))
//     images = imagesFromServer.map(image => ({
//       src: image.url,
//       thumbnail: image.url,
//       thumbnailHeight: image.height,
//       thumbnailWidth: image.width
//     }))

//   this.setState({ images })
// }

// componentWillReceiveProps = (nextProps, nextState) => {
//   if (
//     check(nextProps.images) &&
//     Array.isArray(nextProps.images) &&
//     (check(this.state.images) && Array.isArray(this.state.images))
//   ) {
//     if (!deepEquals(nextProps.images, this.state.images.length)) {
//       // if (nextProps.images.length !== this.state.images.length) {
//       let { images } = nextProps
//       images = images.map(image => ({
//         src: image.url,
//         thumbnail: image.url,
//         thumbnailHeight: image.height,
//         thumbnailWidth: image.width
//       }))

//       this.setState({ images })
//     }
//   }
// }

// render = () => {
//   const {
//     category,
//     categories,
//     images: imagesFromServer,
//     filters
//   } = this.props
//   const { images } = this.state
//   const alive = check(imagesFromServer)
//   return (
//     <main className="dashboard-page">
//       {
//         <div>
//           <section className="image-results">
//             {alive &&
//               imagesFromServer.length === 0 && (
//                 <p className="null-case">No images to display.</p>
//               )}
//             {alive &&
//               imagesFromServer.length > 0 && (
//                 <ReactGridGallery
//                   images={check(imagesFromServer) ? images : []}
//                   enableImageSelection={false}
//                   onClickThumbnail={this.handleThumbnailClick}
//                 />
//               )}
//           </section>
//           <aside className="primary-config">
//             {category === 'art' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">medium</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['medium']}
//                     data-filter-name="medium"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('medium') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">device</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['device']}
//                     data-filter-name="device"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('device') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">project</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['project']}
//                     data-filter-name="project"
//                     onChange={this.handleFilterChange}
//                   >
//                     {categories !== undefined &&
//                     categories.hasOwnProperty('device') ? (
//                       categories['medium'].map(option => (
//                         <option key={option.name} value={option.name}>
//                           {option.name}
//                         </option>
//                       ))
//                     ) : (
//                       <option value="n/a">n/a</option>
//                     )}
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'comments' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">topic</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['topic']}
//                     data-filter-name="topic"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">style</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['style']}
//                     data-filter-name="style"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">site</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['site']}
//                     data-filter-name="site"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'clothes' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">wardrobe</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['wardrobe']}
//                     data-filter-name="wardrobe"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">attire</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['attire']}
//                     data-filter-name="attire"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//             {category === 'people' && (
//               <main>
//                 <form className="filter-container">
//                   <label htmlFor="select">mood</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['mood']}
//                     data-filter-name="mood"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">group</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['group']}
//                     data-filter-name="group"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//                 <form className="filter-container">
//                   <label htmlFor="select">angle</label>
//                   <div className="angle-down">
//                     <FontAwesomeIcon icon="angle-down" className="icon" />
//                   </div>
//                   <select
//                     name="select"
//                     className="select"
//                     value={filters['angle']}
//                     data-filter-name="angle"
//                     onChange={this.handleFilterChange}
//                   >
//                     <option value="art">art</option>
//                     <option value="comments">comments</option>
//                     <option value="clothes">clothes</option>
//                     <option value="people">people</option>
//                   </select>
//                 </form>
//               </main>
//             )}
//           </aside>
//           <aside className="secondary-config">
//             {category === 'art' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="color">color</label>
//                     <input
//                       className="input"
//                       name="color"
//                       type="text"
//                       placeholder="#hex or name"
//                     />
//                     <div className="color-group">
//                       <div className="color-1" />
//                       <div className="color-2" />
//                       <div className="color-3" />
//                       <FontAwesomeIcon icon="angle-right" className="icon" />
//                     </div>
//                   </div>
//                 </div>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="length">length</label>
//                     <input name="length" type="text" placeholder="px" />
//                   </div>
//                   <div className="input-group">
//                     <label htmlFor="width">width</label>
//                     <input name="width" type="text" placeholder="px" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'comments' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="username">username</label>
//                     <input className="names" name="username" type="text" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'clothes' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="color">color</label>
//                     <input
//                       className="input"
//                       name="color"
//                       type="text"
//                       placeholder="#hex or name"
//                     />
//                     <div className="color-group">
//                       <div className="color-1" />
//                       <div className="color-2" />
//                       <div className="color-3" />
//                       <FontAwesomeIcon icon="angle-right" className="icon" />
//                     </div>
//                   </div>
//                 </div>
//                 <div className="filter-group gender">
//                   <div className="input-group">
//                     <label>gender</label>
//                     <FontAwesomeIcon icon="male" className="male-icon" />
//                   </div>
//                   <div className="input-group">
//                     <FontAwesomeIcon icon="female" className="female-icon" />
//                   </div>
//                 </div>
//               </main>
//             )}
//             {category === 'people' && (
//               <main>
//                 <div className="filter-group">
//                   <div className="input-group">
//                     <label htmlFor="name">name</label>
//                     <input className="names" name="name" type="text" />
//                   </div>
//                 </div>
//                 <div className="filter-group gender">
//                   <div className="input-group">
//                     <label>gender</label>
//                     <FontAwesomeIcon icon="male" className="male-icon" />
//                   </div>
//                   <div className="input-group">
//                     <FontAwesomeIcon icon="female" className="female-icon" />
//                   </div>
//                 </div>
//               </main>
//             )}
//           </aside>
//         </div>
//       }
//     </main>
//   )
// }
