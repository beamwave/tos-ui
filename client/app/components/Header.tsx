import React, { Component, createRef } from 'react'
import { Link, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap'
import { throttle, debounce } from 'throttle-debounce'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import {
  sidebar,
  toggleEditMode,
  changeCategory,
  trainingWheelsProtocol
} from '../actions/app'
import { query, startSearch } from '../actions/filter'
import { startLogout } from '../actions/auth'
import { loadModal } from '../actions/modal'
import { ADD_IMAGE_MODAL } from '../constants/modaltypes'

interface RProps {
  photo?: string
  username?: string
  isAuthenticated?: boolean
  trainingWheels?: boolean
  query?: any
  input?: string
  category?: string
  backButton?: boolean
  history?: any
}

interface RState {
  sidebar?: any
  search?: any
  startSearch?: (query: any) => any
  toggleEditMode?: () => any
  trainingWheelsProtocol?: () => any
  changeCategory?: (string) => any
  loadModal?: (string) => any
  startLogout?: () => void
}

type Props = RProps & RState

export class Header extends Component<Props> {
  state = {
    // data: {
    //   query: '',
    //   category: 'art'
    //   // category: this.props.category && this.props.category
    // },
    input: '',
    dropdownOpen: false
    // showSearch: false,
    // editMode: false
  }

  toggleDropdown = () =>
    this.setState({ dropdownOpen: !this.state.dropdownOpen })

  toggleTrainingWheels = () => this.props.trainingWheelsProtocol()

  toggleEditMode = () => this.props.toggleEditMode()

  // needs to be attached to document for click detection
  // componentDidMount = () => {
  //   const input = this.inputRef.current
  //   if (input) input.focus()

  //   document.addEventListener('click', this.determineClick, false)
  // }

  // needs to be attached to document for click detection
  // componentWillUnmount = () =>
  //   document.removeEventListener('click', this.determineClick)

  // addImageModal = () => this.props.loadModal(ADD_IMAGE_MODAL)

  // handleCategoryChange = ({ target }) => {
  //   this.props.changeCategory(target.value)
  // }

  // determineClick = e => {
  //   // if click outside modal or not typing in input, hide search
  //   if (!e.target.closest('.section') && !e.target.dataset.search) {
  //     this.hideSearch()
  //   }

  //   // else if you click inside or type in input, ...
  // }

  // showSearch = () => this.setState({ showSearch: true })

  // hideSearch = () => this.setState({ showSearch: false })

  // private handleCall = async data => this.props.startSearch(data)

  // // should debounce action creator in handleCall for 5 seconds
  // private startSearchDebounce = debounce(1000, this.handleCall)
  // private startSearchThrottle = throttle(750, this.handleCall)

  // private handleText = input => this.setState({ input })

  // private handleSearch = async e => {
  //   e.preventDefault()
  //   // value from onChange
  //   const {
  //     target: { value: input }
  //   } = e
  //   const { category, startSearch } = this.props

  //   this.handleText(input)

  //   // data sent to action creator
  //   const data = {
  //     input,
  //     tags: input.replace(/\s/g, '').split(','),
  //     category: category
  //   }

  //   if (input.length < 5) {
  //     this.startSearchThrottle(data)
  //   } else {
  //     this.startSearchDebounce(data)
  //   }
  // }

  // handleSelectColor = category => {
  //   if (category === 'art') return 'select teal'
  //   if (category === 'comments') return 'select violet'
  //   if (category === 'clothes') return 'select blue'
  //   if (category === 'people') return 'select grey'
  // }

  private inputRef = React.createRef<HTMLInputElement>()

  render = () => {
    const {
      username,
      isAuthenticated,
      startLogout,
      history
      // photo,
      // trainingWheels,
      // backButton,
    } = this.props
    // const {
    //   input,
    //   data: { category }
    // } = this.state

    return (
      <header className="global-header">
        <p className="logo">Collegeboard</p>
        <div className="user">
          <p>{username}</p>
          <div className="profile-image" />
        </div>
      </header>
      // <header className="nav-header">
      //   {!backButton ? (
      //     <div className="plus" onClick={this.addImageModal}>
      //       <FontAwesomeIcon icon="plus" />
      //       <p className="hint">upload image</p>
      //     </div>
      //   ) : (
      //     <div className="plus" onClick={() => history.goBack()}>
      //       <FontAwesomeIcon icon="arrow-left" />
      //       <p className="hint">go back</p>
      //     </div>
      //   )}
      //   <div className="settings">
      //     <FontAwesomeIcon icon="cog" />
      //     <p className="hint">settings</p>
      //   </div>
      //   {!backButton && (
      //     <div className="protocol" onClick={this.toggleTrainingWheels}>
      //       {trainingWheels ? (
      //         <FontAwesomeIcon icon="lock" />
      //       ) : (
      //         <FontAwesomeIcon icon="unlock" />
      //       )}
      //       <p className="hint">
      //         {trainingWheels ? 'deactive' : 'active'} training wheels protocol
      //         )}
      //       </p>
      //     </div>
      //   )}
      //   {backButton && (
      //     <div className="protocol" onClick={this.toggleEditMode}>
      //       <FontAwesomeIcon icon="sliders-h" />
      //       <p className="hint">adjust image settings</p>
      //     </div>
      //   )}
      //   <div className="search-group">
      //     <FontAwesomeIcon icon="search" className="icon" />
      //     <form className="submit" onSubmit={this.handleSearch}>
      //       <input
      //         type="text"
      //         className="input spawnSearch"
      //         placeholder="Search"
      //         onChange={this.handleSearch}
      //         value={input}
      //         ref={this.inputRef}
      //       />

      //       <div className="type-choice">
      //         <div className="angle-down">
      //           <FontAwesomeIcon icon="angle-down" className="icon" />
      //         </div>
      //         <select
      //           className={this.handleSelectColor(category)}
      //           value={category}
      //           onChange={this.handleCategoryChange}
      //         >
      //           <option value="art">art</option>
      //           <option value="comments">comments</option>
      //           <option value="clothes">clothes</option>
      //           <option value="people">people</option>
      //           <option value="music">music</option>
      //           <option value="quora">quora</option>
      //           <option value="bookmarks">bookmarks</option>
      //           <option value="recipes">recipes</option>
      //           <option value="vocabulary">vocabulary</option>
      //           <option value="notes">notes</option>
      //           <option value="timeline">timeline</option>
      //         </select>
      //       </div>
      //       <div className="search-options">
      //         <p className="save-search">save search</p>
      //         <p className="advanced-search">advanced search</p>
      //       </div>
      //     </form>
      //   </div>

      //   <Dropdown
      //     isOpen={this.state.dropdownOpen}
      //     toggle={this.toggleDropdown}
      //     className="dropdown-root"
      //   >
      //     <DropdownToggle className="dropdown-toggle">
      //       <p className="nav-username">{username}</p>
      //       <div
      //         className="image"
      //         style={{
      //           // url bg-pos: x% y% / bg-scale: %
      //           background: `url(${photo}) 50% 5% / 110% no-repeat`,
      //           marginRight: 15
      //         }}
      //       />
      //       <FontAwesomeIcon icon="angle-down" className="icon fa-angle-down" />
      //     </DropdownToggle>
      //     <DropdownMenu
      //       right
      //       className="dropdown-menu"
      //       style={{
      //         display: this.state.dropdownOpen === false ? 'none' : 'block'
      //       }}
      //     >
      //       <DropdownItem className="dropdown-item">
      //         <Link to="/profile" className="profile">
      //           Profile
      //         </Link>
      //       </DropdownItem>
      //       <DropdownItem
      //         className="dropdown-item"
      //         onClick={() => startLogout()}
      //       >
      //         {isAuthenticated ? <p className="logout">Logout</p> : null}
      //       </DropdownItem>
      //     </DropdownMenu>
      //   </Dropdown>
      // </header>
    )
  }
}

const mapStateToProps = (state, props) => ({
  // photo: state.auth.profileImage,
  username: state.auth.username
  // isAuthenticated: !!state.auth.token,
  // category: state.app.category,
  // trainingWheels: state.app.trainingWheels,
  // backButton: props.match.params._id ? true : false // for some reason this works
})

export default withRouter<any>(
  connect<RProps, any>(
    mapStateToProps,
    {
      sidebar,
      query,
      startSearch,
      toggleEditMode,
      changeCategory,
      trainingWheelsProtocol,
      loadModal,
      startLogout
    }
  )(Header)
)
