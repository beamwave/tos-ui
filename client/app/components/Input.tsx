import React, { Component } from 'react'

const Input = ({ query, handleChange }) => {
  const Input = (
    <input
      type="text"
      className="input spawnSearch"
      placeholder="Search"
      onChange={handleChange}
      value={query}
    />
  )

  return Input
}

export default Input

// import React, { Component } from 'react'

// interface IProps {
//   handleSearch: (e: any) => Promise<void>
// }

// export class Input extends Component<IProps> {
//   state = {
//     query: ''
//   }

//   handleChange = e => {
//     e.preventDefault()
//     const {
//       target: { value: query }
//     } = e
//     this.setState({ query })
//     this.props.handleSearch(e)
//   }

//   shouldComponentUpdate = (nextProps, nextState) => {
//     console.log('next state', this.state.query)
//     console.log('incoming query', nextState.query)

//     if (nextState.query !== this.state.query) return true

//     return false
//   }
//   render = () => {
//     const { query } = this.state
//     return (
//       <input
//         type="text"
//         className="input spawnSearch"
//         placeholder="Search"
//         onChange={this.handleChange}
//         value={query}
//       />
//     )
//   }
// }

// export default Input
