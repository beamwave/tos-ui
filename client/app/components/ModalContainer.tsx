import React from 'react'
import { connect } from 'react-redux'

// import all modal components
import AddImageModal from './modals/AddImageModal'
import gradeAdjustmentsModal from './modals/GradeAdjustmentsModal'

// import modal type constants
import {
  ADD_IMAGE_MODAL,
  GRADE_ADJUSTMENTS_MODAL
} from '../constants/modaltypes'

// modal directory based on props.modalType
const MODAL_COMPONENTS = {
  ADD_IMAGE_MODAL: AddImageModal,
  GRADE_ADJUSTMENTS_MODAL: gradeAdjustmentsModal
}

export const ModalContainer = props => {
  // if no modal set in store, do not render one
  if (!props.modalType) {
    return null
  }

  // object lookup
  const SpecificModal = MODAL_COMPONENTS[props.modalType]

  return <SpecificModal />
}

const mapStateToProps = state => ({
  modalType: state.modal.modalType
})

export default connect(mapStateToProps)(ModalContainer)
