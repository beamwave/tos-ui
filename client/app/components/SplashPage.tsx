import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link, Redirect } from 'react-router-dom'

export class SplashPage extends Component {
  // @ts-ignore
  render = () => {
    return (
      <div className="splash-page">
        <header>
          <nav className="splash-nav">
            <Link to="/" className="item">
              Arsenal
            </Link>
            <Link to="/login" className="item">
              Login
            </Link>
            <p>or</p>
            <Link to="/signup" className="item">
              Signup
            </Link>
          </nav>
        </header>
        <main>
          <p>splash page</p>
        </main>
      </div>
    )
  }
}

export default connect()(SplashPage)
