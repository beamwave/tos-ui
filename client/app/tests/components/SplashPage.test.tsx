import React, { Component } from 'react'
import { shallow } from 'enzyme'
import { SplashPage } from '../../components/SplashPage'

test('should render SplashPage component', () => {
  const wrapper = shallow(<SplashPage />)
  expect(wrapper).toMatchSnapshot()
})
