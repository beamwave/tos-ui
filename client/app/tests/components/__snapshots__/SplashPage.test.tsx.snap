// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SplashPage component 1`] = `
<div
  className="splash-page"
>
  <header>
    <nav
      className="splash-nav"
    >
      <Link
        className="item"
        replace={false}
        to="/"
      >
        Arsenal
      </Link>
      <Link
        className="item"
        replace={false}
        to="/login"
      >
        Login
      </Link>
      <p>
        or
      </p>
      <Link
        className="item"
        replace={false}
        to="/signup"
      >
        Signup
      </Link>
    </nav>
  </header>
  <main>
    <p>
      splash page
    </p>
  </main>
</div>
`;
